### DISCUSSION TITLE
Phlex 1.2 -- blocks?

### DISCUSSION THREAD

**Discussion by Bumppoman** [yippee-fun/phlex#454] [Help] [answered]
Hi,

Given the following which worked as-is in < 1.2:

```ruby
# frozen_string_literal: true

module Views
  module Components
    class DataTable::Row::Cell < Base
      attr_reader :condensed

      def initialize(**options, &content)
        @condensed = options.delete(:condensed) { false }
        @content = content
        super(**options)
      end

      def template
        return if options[:show] == false

        td(
          **mix(
            options,
            classes(
              'whitespace-nowrap py-4 pl-4 pr-3 text-sm font-normal text-slate-500 first:pl-8 dark:text-slate-300',
              -> { condensed == :hidden } => 'group-condensed:hidden',
              -> { condensed == :only } => 'hidden group-condensed:table-cell'
            )
          ),
          &@content
        )
      end
    end
  end
end
```

Now that Phlex is taking my block, what is the recommended way to do this?  I know (learned through trial and error, rather) I can make it instantly work by getting rid of the block argument in the initializer and replacing `&@content` with `&@_content_block` but this obviously seems like a private API I'd rather not rely on.

Thanks!

**Comment by joeldrapper** [comment, marked as answer]
The content block will be given to `template` so you can either `yield` inside `template` or update it to take a block, e.g. `def template(&content)`.

**Reply by Bumppoman** [reply]
Ha!  I tried everything else but that.  I was thinking the template method was getting the block magically in the new update and wondering why I couldn't get the right recipe.  Thanks as always!

**Reply by joeldrapper** [reply]
No problem!
